File: java/src/test/java/mx/iteso/GoogleDocAdapterTest.java
package mx.iteso;

import mx.iteso.adapter.*;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

public class GoogleDocAdapterTest {
    GoogleAdapter googleDocAdapter;
    GoogleDoc mockedGoogleDoc;

    @Before
    public void setUp(){
        mockedGoogleDoc = new GoogleDoc(new Font("Arial"), new BackgroundImage(new Image("imageUrl")),new Format(new Font("Arial"), "Regular"),1);
        googleDocAdapter = new GoogleAdapter(mockedGoogleDoc);
    }

    @Test
    public void getFormatTest() {
        String result = googleDocAdapter.getFormat().getFont();
        Assert.assertEquals("Arial", result);
    }

    @Test
    public void getBackgroundTest() {
        String result = googleDocAdapter.getBackground().getUrl();
        Assert.assertEquals("imageUrl", result);

    }

    @Test
    public void setMSOfficeVersionTest() {
        googleDocAdapter.setMSOfficeVersion(3);
        Assert.assertEquals(3, googleDocAdapter.getMSOfficeVersion(), 0);
    }

    @Test
    public void getLicenseTest() {
        String result = googleDocAdapter.getLicense().getLicense();
        Assert.assertEquals("Google license", result);
    }

    @Test
    public void restrictEditIfLicenseIsInvalidTestFalse() {
        MSLicense ms = new MSLicense("Ms");
        boolean result = googleDocAdapter.restrictEditIfLicenseIsInvalid(ms);
        Assert.assertEquals(false, result);

    }
    @Test
    public void restrictEditIfLicenseIsInvalidTestTrue() {
        MSLicense ms = new MSLicense("Google license");
        boolean result = googleDocAdapter.restrictEditIfLicenseIsInvalid(ms);
        Assert.assertEquals(true, result);
    }
}

File: csharp/UnitTestAdapter/GoogleDocAdapterTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Adapter.Clases;

namespace UnitTestAdapter
{
    [TestClass]
    public class GoogleDocAdapterTest
    {
        private GoogleAdapter _googleDocAdapter;
        private GoogleDoc _mockedGoogleDoc;

        [TestInitialize]
        public void SetUp()
        {
            _mockedGoogleDoc = new GoogleDoc(new Font("Arial"), new BackgroundImage(new Image("imageUrl")), new Format(new Font("Arial"), "Regular"), 1);
            _googleDocAdapter = new GoogleAdapter(_mockedGoogleDoc);
        }

        [TestMethod]
        public void GetFormatTest()
        {
            var result = _googleDocAdapter.GetFormat().GetFont();
            Assert.AreEqual("Arial", result);
        }

        [TestMethod]
        public void GetBackgroundTest()
        {
            var result = _googleDocAdapter.GetBackground().GetUrl();
            Assert.AreEqual("imageUrl", result);

        }

        [TestMethod]
        public void SetMsOfficeVersionTest()
        {
            _googleDocAdapter.SetMsOfficeVersion(3);
            Assert.AreEqual(3, _googleDocAdapter.GetMsOfficeVersion(), 0);
        }

        [TestMethod]
        public void GetLicenseTest()
        {
            var result = _googleDocAdapter.GetLicense().GetLicense();
            Assert.AreEqual("Google license", result);
        }

        [TestMethod]
        public void RestrictEditIfLicenseIsInvalidTestFalse()
        {
            var ms = new MsLicense("Ms");
            var result = _googleDocAdapter.RestrictEditIfLicenseIsInvalid(ms);
            Assert.AreEqual(false, result);
        }
        [TestMethod]
        public void RestrictEditIfLicenseIsInvalidTestTrue()
        {
            var ms = new MsLicense("Google license");
            var result = _googleDocAdapter.RestrictEditIfLicenseIsInvalid(ms);
            Assert.AreEqual(true, result);

        }
    }
}
